wallet-next/src/service.rs

```diff
@@ -114,7 +119,34 @@ impl WalletProtocol for WalletService {
         self.check_worker().await?;
         self.check_fvk(request.get_ref().fvk_hash.as_ref()).await?;
 
-        todo!()
+        let include_spent = request.get_ref().include_spent;
+        let asset_id = request
+            .get_ref()
+            .asset_id
+            .to_owned()
+            .map(|x| asset::Id::try_from(x).expect("AssetId protobuf conversion error"));
+        let diversifier_index = request.get_ref().diversifier_index.to_owned().map(|x| {
+            DiversifierIndex::try_from(x).expect("DiversifierIndex protobuf conversion error")
+        });
+        let amount_to_spend = request.get_ref().amount_to_spend;
+
+        let notes = self
+            .storage
+            .notes(include_spent, asset_id, diversifier_index, amount_to_spend)
+            .await
+            .map_err(|_| tonic::Status::unavailable("database error"))?;
+
+        let stream = try_stream! {
```

Comment:
Should the `storage` method be refactored to allow iteration over a range of heights, which would make returning a stream here make more sense? Or should we change the method signature in the service to return a `Vec<NoteRecord>`? Leaving as-is for now, but seems like one side or the other should change.

Reply:
Seems like if we leave it as is, we have the option of changing just the wallet impl to steam data at a later point without having to change anything on the client side